pkg/gencmd/main.go

```diff
@@ -0,0 +1,136 @@
+package gencmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type Cmd interface {
+	Command() *cobra.Command
+	Execute() error
+	MustExecute()
+}
+
+type cmdImpl struct {
+	cobraCmd *cobra.Command
+	initErr  error
+}
+
+func (c *cmdImpl) Command() *cobra.Command {
+	return c.cobraCmd
+}
+
+func (c *cmdImpl) Execute() error {
+	if c.initErr != nil {
+		return errors.WithStack(c.initErr)
+	}
+	return errors.WithStack(c.cobraCmd.Execute())
+}
+
+func (c *cmdImpl) MustExecute() {
+	var code int
+
+	if err := c.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		code = 1
+	}
+
+	os.Exit(code)
+}
+
+func New(
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function New should have comment or be unexported